### ISSUE TITLE
OutputDebugStringW Unicode display issue

### ISSUE THREAD

**Issue by alpan** [x13pixels/remedybg-issues#214]
Hi,

RemedyBG seems to be having some trouble properly displaying Unicode strings in the Output window. Below is an image where the character "ç" is retrieved from a Windows WM_CHAR message. The Watch window has no trouble showing the correct string.

VS 2017 displays the correct string in its Output window.

![image](https://user-images.githubusercontent.com/679222/115175724-37866e80-a0d4-11eb-9996-471d747a2158.png)

Sorry if this report seems a bit sparse. Let me know if you'd like more information or clarification, and thank you for your work on the debugger.

**Comment by Feacur** [comment]
@alpan, a minor [unrealated and nitpicky, sorry for that,] addition: don't you need to account for surrogate pairs?
```
// example~~~~~
case WM_CHAR: {
	uint32_t value = (uint32_t)wParam;

	static uint32_t utf16_high_surrogate = 0;
	if ((0xd800 <= value) && (value <= 0xdbff)) {
		// if (utf16_high_surrogate != 0) { DEBUG_BREAK(); utf16_high_surrogate = 0; return 0; }
		utf16_high_surrogate = value;
		return 0;
	}

	if ((0xdc00 <= value) && (value <= 0xdfff)) {
		// if (utf16_high_surrogate == 0) { DEBUG_BREAK(); return 0; }
		value = (((utf16_high_surrogate & 0x03ff) << 10) | (value & 0x03ff)) + 0x10000;
		utf16_high_surrogate = 0;
	}

	input_to_platform_on_codepoint(value);
	return 0;
}
```

`WM_CHAR` gets you UTF-16 encoded data (given you created window class and window itself and, optionally, supplied `-DUNICODE` define flag, of course); thus actual UTF-32 codepoint might be decoded either from a single message or from a consequent pair of messages:
(https://www.ietf.org/rfc/rfc2781.txt)
(https://docs.microsoft.com/en-us/windows/win32/inputdev/wm-char)
(https://docs.microsoft.com/en-us/windows/win32/intl/surrogates-and-supplementary-characters)
(https://handmade.network/forums/t/2011-keyboard_inputs_-_scancodes,_raw_input,_text_input,_key_names)

P.S.: I wasn't able to get a single `WM_UNICHAR` message; that's why I resorted to the universal method
(here's UTF-8 spec if you ever need it, too: https://www.ietf.org/rfc/rfc3629.txt)

**Comment by alpan** [comment]
Hi @Feacur, you are probably correct: I have just begun writing a platform layer to gain technical experience in Windows and graphics programming with zero prior knowledge (and although I'm quite well acquainted with Handmade Hero as you might expect, I'm deliberately trying to avoid referring to it as a crutch). So it may well be that your point becomes salient in the future. Thanks for the heads-up.

**Comment by x13pixels** [comment]
[Fixed in upcoming patch release 0.3.5.1] We now "opt in" to receive Unicode strings output with OutputDebugStringW in Windows 10 and later by using WaitForDebugEventEx rather than WaitForDebugEvent in the debug loop.
